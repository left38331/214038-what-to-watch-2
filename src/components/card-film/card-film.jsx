import React from 'react';
import PropTypes from 'prop-types';

import {VideoPlayer} from 'components/video-player/video-player';

export const CardFilm = (props) => {
  return <article className="small-movie-card catalog__movies-card" onMouseEnter={()=> props.hoverCardHandler(props.film)} onMouseLeave={props.leaveCardHandler}>
    <div className="small-movie-card__image">
      <VideoPlayer
        poster={props.film.poster}
        preview={props.film.preview}
        isPlaying={props.isPlaying}
        muted={true}
      />
    </div>
    <h3 className="small-movie-card__title">
      <a className="small-movie-card__link" href="#" onClick={props.clickTitle}>{props.film.title}</a>
    </h3>
  </article>;
};

CardFilm.propTypes = {
  film: PropTypes.shape({
    title: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    preview: PropTypes.string.isRequired,
  }),
  clickTitle: PropTypes.func.isRequired,
  hoverCardHandler: PropTypes.func.isRequired,
  leaveCardHandler: PropTypes.func.isRequired,
  isPlaying: PropTypes.bool.isRequired
};
