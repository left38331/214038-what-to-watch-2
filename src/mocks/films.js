export const films = [
  {
    id: 1,
    title: `Fantastic Beats`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    alt: `Fantastic Beasts: The Crimes of Grindelwald`,
    link: `movie-page.html`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/trailers/fbwft-rutlr3_854.mp4`,
    genre: `Fantasy`
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    poster: `img/bohemian-rhapsody.jpg`,
    alt: `Bohemian Rhapsody`,
    link: `movie-page.html`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/c/76/c766c75666111cab4aa6bb75a5424bc0.480.mp4`,
    genre: `Drama`
  },
  {
    id: 3,
    title: `Macbeth.jpg`,
    poster: `img/macbeth.jpg`,
    alt: `Macbeth`,
    link: `movie-page.html`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Drama`
  },
  {
    id: 4,
    title: `Aviator`,
    poster: `img/aviator.jpg`,
    alt: `Aviator`,
    link: `movie-page.html`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `Adventure`
  },
  {
    id: 5,
    title: `War of the worlds`,
    poster: `img/war-of-the-worlds.jpg`,
    alt: `War of the worlds`,
    link: `movie-page.html`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/5/2a/52ab2988c90e81715008ed5189632b8d.480.mp4`,
    genre: `Action`
  },
];
