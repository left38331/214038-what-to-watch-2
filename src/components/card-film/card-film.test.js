import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import {CardFilm} from 'components/card-film/card-film';
import {films} from '../../mocks/films';

function createNodeMock(element) {
  if (element.type === `video`) {
    return {
      createRef() {}
    };
  }
  return null;
}

it(`Render correctly card-film component`, () => {
  const options = {createNodeMock};
  const cardFilmComponent = renderer.create(<BrowserRouter><CardFilm
    film = {films[0]}
    isPlaying = {false}
    clickTitle = {()=>{}}
    hoverCardHandler = {()=>{}}
    leaveCardHandler = {()=>{}}
  /></BrowserRouter>, options).toJSON();

  expect(cardFilmComponent).toMatchSnapshot();
});
