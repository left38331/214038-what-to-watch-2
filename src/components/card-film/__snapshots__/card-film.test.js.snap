// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly card-film component 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      src="https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/trailers/fbwft-rutlr3_854.mp4"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      Fantastic Beats
    </a>
  </h3>
</article>
`;
